flask-server.py

```diff
@@ -25,122 +26,167 @@
+    """basic auth for api"""
+
     @wraps(f)
     def decorated_function(*args, **kwargs):
         auth = frequest.authorization
         if not auth or not check_auth(auth.username, auth.password):
-            return jsonify({'message': 'Authentication required'}), 401
+            return jsonify({"message": "Authentication required"}), 401
         return f(*args, **kwargs)
+
     return decorated_function
 
-@app.route('/v0/public.json', methods = ['GET'])  
-def public_json():
 
+@app.route("/v0/public.json", methods=["GET"])
+def public_json():
     return jsonify(alldata)
 
-@app.route('/v0/public.ics')
-def public_ics():    
-    # Create a new calendar
+
+@app.route("/v0/public.ics")
+def public_ics():
     cal = Calendar()
 
-    # Add events to the calendar
     for event in alldata:
         try:
             cal_event = Event()
-            cal_event.add('summary', event['title'])
-            cal_event.add('dtstart', event['start'])
-            cal_event.add('dtend', event['end'])
-            cal_event.add('description', event['description'])
-            
-            location = event['location']
-            location_str = ', '.join(filter(None, [
-                location.get('building'),
-                location.get('room_number'),
-                location.get('street'),
-                location.get('city'),
-                location.get('state'),
-                location.get('zipcode')
-            ]))
-            cal_event.add('location', location_str)
-            cal.add_component(cal_event)
+            cal_event.name = event["title"]
+            cal_event.begin = event["start"]
+            cal_event.end = event["end"]
+            cal_event.description = event["description"]
+
+            location = event["location"]
+            location_str = ", ".join(
+                filter(
+                    None,
+                    [
+                        location.get("building"),
+                        location.get("room_number"),
+                        location.get("street"),
+                        location.get("city"),
+                        location.get("state"),
+                        location.get("zipcode"),
+                    ],
+                )
+            )
+            if location_str:
+                cal_event.location = location_str
+
+            cal.events.add(cal_event)
         except Exception as e:
             print(e)
-        
-    # Create a response object
-    response = Response(cal.to_ical(), mimetype='text/calendar')
-    response.headers['Content-Disposition'] = 'attachment; filename=calendar.ics'
-    
+
+    response = Response(str(cal), mimetype="text/calendar")
+    response.headers["Content-Disposition"] = "attachment; filename=calendar.ics"
     return response
 
+
+@app.route("/v0/public.ics/<event_id>")
+def get_event(event_id):
+    try:
+        # Find first event with matching id
+        event = next(e for e in alldata if e["id"] == event_id)
```

Comment:
Is there a particular place this `id` value is added to the event? i'm a little hesitant to have these values on display to users if they arent coming from some source that has some checks to make sure these ID's are consistent, sensible, or at least URL-safe